### PR TITLE
[CALCITE-5306] Support LTS JDKs and latest in testing

### PR DESCRIPTION
**Pull request by asolimando**
Dropped JDK 15 and 16 by adopting the latest LTS version (JDK 17) or latest overall (JDK18). JDK19 is at the moment incompatible with `forbiddenAPIs`, so it's not an option at the moment.

### CODE REVIEW

**Review by zabetak — APPROVED** [review]
LGTM, thanks for stabilizing the CI.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -123,7 +123,7 @@ jobs:
 
   linux-openj9:
```
Comment (michaelmior) [inline .github/workflows/main.yml:124]: It seems there is still some inconsistency here. I'm not sure if the original intent was to run on macOS or Linux, but I think the key here, the name, and the actual OS used should all match.

Comment (rubenada) [reply .github/workflows/main.yml]: Just to provide some info. IIRC correctly, this was added due to some strange Error that appeared only on WebSphere (ie with IBM's JVM), see details on email discussion: https://lists.apache.org/thread/qmo39634fpdxnwconoo80zbczo0wy1wq

Comment (asolimando) [reply .github/workflows/main.yml]: Thanks @rubenada for pointing to the original discussion. Did you experience the issue described there on a linux server I guess, would you recommend switching to a linux-based image?

If we are not covering the specific use-case on the original OS we might even consider removing that job altogether and focus on more recent JDK versions or reduce the load on CI.

Comment (rubenada) [reply .github/workflows/main.yml]: @asolimando  IIRC the problem appeared on an Ubutu + WebSphere environment.
I agree, IMO we could consider removing this job and, and you say, focus on recent JDKs.

**Pull request merged**